src/worker.py

```
@@ -6536,12 +6096,61 @@ async def on_fetch(request, env, ctx=None) -> Response:
+    # Requires ADMIN_SECRET env variable.
+    # POST body: {"updates": [{"github_username": "mentorX", "referred_by": "contributorY"}, ...]}
+    if method == "POST" and path in {"/admin/backfill-referrer", f"{_admin_path(env)}/backfill-referrer"}:
+        admin_secret = getattr(env, "ADMIN_SECRET", "")
+        if not admin_secret:
+            return _json({"error": "Admin endpoint not configured"}, 403)
+        auth_header = (request.headers.get("Authorization") or "").strip()
+        if auth_header != f"Bearer {admin_secret}":
+            return _json({"error": "Unauthorized"}, 401)
+        try:
+            body = json.loads(await request.text())
+        except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)